agent/rpc/middleware/interceptors.go

```diff
@@ -49,7 +50,11 @@ func NewRequestRecorder(logger hclog.Logger, isLeader func() bool, localDC strin
 }
 
 func (r *RequestRecorder) Record(requestName string, rpcType string, start time.Time, request interface{}, respErrored bool) {
-	elapsed := time.Since(start).Milliseconds()
+	elapsedNs := time.Since(start)
```

Comment:
I think it would be cleaner to start with `time.Since(start).Microseconds()` and convert it to a float64 milisecond rather than "if milliseconds rounds to 0, then use microseconds".

Reply:
Thanks for the feedback. Your suggestions is easier to understand and requires less code (If I actually understood it and implemented it correctly).